xunit-autolabeler-v2/ast_parser/python/source_parser_test.py

```diff
@@ -0,0 +1,236 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import unittest
+from unittest.mock import MagicMock
+
+
+from . import source_parser
+
+import pytest
+
+
+class GetMethodChildrenTest(unittest.TestCase):
+    def _clear_fake_method(self):
+        fake_method = MagicMock(autospec=False)
+
+        del fake_method.id
+        del fake_method.body
+        del fake_method.expr
+        del fake_method.value
+        del fake_method.func
+
+        return fake_method
+
+    def test_handles_id(self):
+        fake_method = self._clear_fake_method()
+        fake_method.id = "abcd"
+
+        results = source_parser._get_method_children(fake_method)
+
+        assert results == ['abcd']
+
+    def test_treats_body_as_list(self):
+        submethod_1 = self._clear_fake_method()
+        submethod_1.id = "submethod_1"
+
+        submethod_2 = self._clear_fake_method()
+        submethod_2.id = "submethod_2"
+
+        fake_method = self._clear_fake_method()
+        fake_method.body = [submethod_1, submethod_2]
+
+        results = source_parser._get_method_children(fake_method)
+
+        assert results == ['submethod_1', 'submethod_2']
+
+    def test_treats_func_as_list(self):
+        submethod_1 = self._clear_fake_method()
+        submethod_1.id = "submethod_1"
+
+        submethod_2 = self._clear_fake_method()
+        submethod_2.id = "submethod_2"
+
+        fake_method = self._clear_fake_method()
+        fake_method.func = [submethod_1, submethod_2]
+
+        results = source_parser._get_method_children(fake_method)
+
+        assert results == ['submethod_1', 'submethod_2']
+
+    def test_treats_func_as_prop(self):
+        submethod = self._clear_fake_method()
+        submethod.id = "submethod"
+
+        fake_method = self._clear_fake_method()
+        fake_method.func = submethod
+
+        results = source_parser._get_method_children(fake_method)
+
+        assert results == ['submethod']
+
+    def test_handles_func_args(self):
+        func_submethod = self._clear_fake_method()
+        func_submethod.id = "submethod_func"
+
+        args_submethod_1 = self._clear_fake_method()
+        args_submethod_1.id = "submethod_args_1"
+
+        args_submethod_2 = self._clear_fake_method()
+        args_submethod_2.id = "submethod_args_2"
+
+        fake_method = self._clear_fake_method()
+        fake_method.body = [
+            func_submethod, args_submethod_1, args_submethod_2]
+
+        results = source_parser._get_method_children(fake_method)
+
+        assert results == [
+            'submethod_func', 'submethod_args_1', 'submethod_args_2']
+
+    def test_body_must_be_list(self):
+        fake_method = self._clear_fake_method()
+        fake_method.body = "not_a_list"
+
+        results = source_parser._get_method_children(fake_method)
+
+        assert results == []
+
+
+class GetEndingLineTest(unittest.TestCase):
```

Comment:
ditto: similar repetitive test that could be way simpler to read with parameterized.

Reply:
`parametrized` tests don't work when wrapped in a class, so I moved these to another file.